From 5476c01fa183f7a9a7fe350f11371fe589cb00f6 Mon Sep 17 00:00:00 2001
From: Petr Lautrbach <plautrba@redhat.com>
Date: Wed, 5 Oct 2016 15:27:57 +0200
Subject: [PATCH 42/43] sandbox: do not try setup directories without -X or -M

sandbox tried to copy all affected files to the new home
or tmp even though -M or -X was not specified and there was no new
directory.

Fixes:
$ sandbox ls ~
/usr/bin/sandbox: [Errno 17] File exists: '/root'

Signed-off-by: Petr Lautrbach <plautrba@redhat.com>
(cherry picked from SELinuxProject commit 5b98f39de5119dcc865f6c7923682fb10219d45a)
---
 sandbox/sandbox | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/sandbox/sandbox b/sandbox/sandbox
index b77b438..f4f6c29 100644
--- a/sandbox/sandbox
+++ b/sandbox/sandbox
@@ -461,7 +461,8 @@ sandbox [-h] [-l level ] [-[X|M] [-H homedir] [-T tempdir]] [-I includefile ] [-
         try:
                self.__parse_options()
                self.__gen_context()
-               self.__setup_dir()
+               if self.__mount:
+                   self.__setup_dir()
                return self.__execute()
         except KeyboardInterrupt:
             sys.exit(0)
-- 
1.8.3.1

